platform/admin/admin-configuration-passwordencryption/src/main/java/org/codice/ddf/admin/configuration/password/encryption/PasswordEncryptingPersistenceManager.java

```diff
@@ -0,0 +1,125 @@
+import org.codice.ddf.admin.core.api.ConfigurationAdmin;
+import org.codice.ddf.admin.core.api.Metatype;
+import org.codice.ddf.admin.core.api.MetatypeAttribute;
+import org.codice.ddf.admin.core.api.Service;
+import org.codice.ddf.security.common.Security;
+import org.osgi.service.metatype.AttributeDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.encryption.EncryptionService;
+
+public class PasswordEncryptingPersistenceManager implements PersistenceManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            PasswordEncryptingPersistenceManager.class);
+
+    private PersistenceManager persistenceManager;
+
+    //findbugs incorrectly flagged this as an unread field and there was no way to skip it with the annotations that worked
+    //findbugs is thus disabled for this class
+    private EncryptionService encryptionService;
+
+    private ConfigurationAdmin configurationAdmin;
+
+    public PasswordEncryptingPersistenceManager(PersistenceManager persistenceManager,
+            EncryptionService encryptionService, ConfigurationAdmin configurationAdmin) {
+        this.persistenceManager = persistenceManager;
+        this.encryptionService = encryptionService;
+        this.configurationAdmin = configurationAdmin;
+    }
+
+    @Override
+    public boolean exists(String pid) {
+        return persistenceManager.exists(pid);
+    }
+
+    @Override
+    public Dictionary load(String pid) throws IOException {
+        Dictionary dictionary = persistenceManager.load(pid);
+        updatePasswords(pid, dictionary, encryptionService::decrypt);
+        return dictionary;
+    }
+
+    @Override
+    public Enumeration getDictionaries() throws IOException {
+        return persistenceManager.getDictionaries();
+    }
+
+    @Override
+    public void store(String pid, Dictionary dictionary) throws IOException {
+        Dictionary tempDictionary = new Hashtable();
+        Enumeration keys = dictionary.keys();
+        while (keys.hasMoreElements()) {
+            Object key = keys.nextElement();
+            tempDictionary.put(key, dictionary.get(key));
+        }
+        updatePasswords(pid, tempDictionary, encryptionService::encrypt);
+        persistenceManager.store(pid, tempDictionary);
+    }
+
+    private void updatePasswords(String pid, Dictionary dictionary,
+            Function<String, String> function) {
+        int indexOfDot = pid.lastIndexOf(".");
+        String trimmedPid = pid.substring(0, (indexOfDot == -1) ? pid.length() - 1 : indexOfDot);
+        List<Service> services = null;
+        try {
+            services = getServices(trimmedPid);
+        } catch (PrivilegedActionException e) {
+            LOGGER.info(
+                    "Unable to find metatype for service call. Passwords in configuration could be unencrypted.",
+                    e);
+        }
+        if (services != null && services.size() == 1) {
```

Comment:
That would be undefined, since you should only ever have a single service for any particular PID. They are _supposed_ to be unique.